src/api/customer/customer-business.ts

```diff
@@ -11,6 +11,38 @@ const persistCustomer: CustomerCreationRequestHandler = async (req, res, next) =
     }
 }
 
+const findAllCustomer: CustomerFindAllRequestHandler = async (req, res, next) => {
+    try {
+        let{ offset, limit } =  res.locals;
+
+        if(!offset) offset = 0;
+        if(!limit) limit = 50;
+
+        res.locals.customersFound = await Customer.findAll({ offset, limit });
+        res.locals.offset = offset;
+        res.locals.limit = limit;
+        next();
+    } catch(error) {
+        next(error);
+    }
+}
+
+const findByIdCustomer: CustomerFindByIdRequestHandler = async (req, res, next) => {
+    try {
+        const { customerUuid  } = res.locals;
+        await Customer.findByPk(customerUuid).then(( result ) => {
+            if(result?.uuid){
+                res.locals.customerFound = result
+            }            
+        });
+        next();
```

Comment:
Aqui houve uma confusão.

Ou se usa o padrão `async-await`, ou usa o encadeamento de `then`, ambos não é possível.
Até poderia ser possível, mas o `then` deveria que ter algum retorno (retornar uma outra promise).

Considere a primeira opção como a melhor (usando `async-await`):
```suggestion
        const customer = await Customer.findByPk(customerUuid);
        res.locals.customerFound = customer;
        next();
```
E essa segunda opção como secundária (usando `.then`), repare que neste caso a chamada de `next` fica do lado de dentro (dentro do corpo da função de callback).
Esse seria o exemplo usando `.then`, mas não é esse o padrão que adotamos desde o início do projeto:
```suggestion
        Customer.findByPk(customerUuid).then(( result ) => {
            if(result?.uuid){
                res.locals.customerFound = result
            }
            next();
        });
```